internal/tiltfile/analytics/analytics.go

```diff
@@ -20,12 +22,26 @@ func NewExtension() Extension {
 
 func (e Extension) NewState() interface{} {
 	return Settings{
-		Opt: analytics.OptDefault,
+		Opt:                analytics.OptDefault,
+		CustomTagsToReport: make(map[string]string),
 	}
 }
 
 func (Extension) OnStart(env *starkit.Environment) error {
-	return env.AddBuiltin("analytics_settings", setAnalyticsSettings)
+	err := env.AddBuiltin("analytics_settings", setAnalyticsSettings)
+	if err != nil {
+		return err
+	}
+
+	// This is an experimental feature to allow Tiltfiles to specify custom data to report to analytics
+	// to allow teams to get more visibility into, e.g., who's using Tilt or what k8s distributions are
+	// their members using. It is not intended for use without coordinating with the Tilt team.
+	err = env.AddBuiltin("experimental_report_custom_tags", reportCustomTags)
```

Comment:
fwiw, i might have tried to squeeze `analytics` into this name somehow to clarify that they're connected? maybe `experimental_analytics_report(tags)`?